tests/problems.py

```diff
@@ -12,35 +12,44 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Problems with analytical solutions."""
+"""Problems of different noise types with analytical solutions.
+
+Ex1, Ex2, Ex3 from
+
+Rackauckas, Christopher, and Qing Nie. "Adaptive methods for stochastic
+differential equations via natural embeddings and rejection sampling with memory."
+Discrete and continuous dynamical systems. Series B 22.7 (2017): 2731.
+
+Ex4 is constructed to test schemes for SDEs with general noise.
+"""
+
 import torch
 from torch import nn
 
 from torchsde import BaseSDE
+from torchsde.settings import NOISE_TYPES, SDE_TYPES
 
 
 class Ex1(BaseSDE):
-    def __init__(self, d=10, sde_type='ito'):
-        super(Ex1, self).__init__(noise_type="diagonal", sde_type=sde_type)
+    def __init__(self, d=10, sde_type=SDE_TYPES.ito):
+        super(Ex1, self).__init__(sde_type=sde_type, noise_type=NOISE_TYPES.diagonal)
         self._nfe = 0
+
         # Use non-exploding initialization.
         sigma = torch.sigmoid(torch.randn(d))
         mu = -sigma ** 2 - torch.sigmoid(torch.randn(d))
         self.mu = nn.Parameter(mu, requires_grad=True)
         self.sigma = nn.Parameter(sigma, requires_grad=True)
 
     def f(self, t, y):
-        del t
         self._nfe += 1
         return self.mu * y
 
     def f_corr(self, t, y):
-        del t
         self._nfe += 1
         return self.mu * y - .5 * (self.sigma ** 2) * y
```

Comment:
None of the `f_corr` are used. I don't think the Strat diagnostics will be correct.
(Easiest thing would be to just assign the correct f in `__init__`)

Reply:
There should not be any serious problems with the diagnostics, since I'm running Euler (for Ito) and Midpoint (for Strat) with very small step size for the ground-truth. This makes the code consistent.

Reply:
Another note is that we should keep `f_corr` just in case we need them in the future.